tests/unit/dataactcore/test_job_queue.py

```diff
@@ -31,13 +25,68 @@ def test_generate_f_file(monkeypatch, mock_broker_config_paths):
     expected = [['key4', 'key11'], ['a', 'b'], ['c', 'd']]
-    assert read_f_file_rows('unique1', file_path) == expected
+    jobQueue.generate_f_file(1, 1, Mock(), 'uniq1', 'uniq1', is_local=True)
+    assert read_file_rows(file_path) == expected
 
     # re-order
     fileF_mock.mappings = OrderedDict(
         [('key11', 'mapping11'), ('key4', 'mapping4')])
-    file_path = str(mock_broker_config_paths['broker_files'].join('unique2'))
+    file_path = str(mock_broker_config_paths['broker_files'].join('uniq2'))
     expected = [['key11', 'key4'], ['b', 'a'], ['d', 'c']]
-    assert read_f_file_rows('unique2', file_path) == expected
+    jobQueue.generate_f_file(1, 1, Mock(), 'uniq2', 'uniq2', is_local=True)
+    assert read_file_rows(file_path) == expected
+
+
+def test_generate_e_file(monkeypatch, mock_broker_config_paths, database):
+    """A CSV with fields in the right order should be written to the file
+    system"""
+    # Generate several file D1 entries, largely with the same submission_id,
+    # and with two overlapping DUNS
+    model = AwardProcurementFactory()
+    aps = [AwardProcurementFactory(submission_id=model.submission_id)
+           for i in range(4)]
+    same_duns = AwardProcurementFactory(
+        submission_id=model.submission_id,
+        awardee_or_recipient_uniqu=model.awardee_or_recipient_uniqu)
+    unrelated = AwardProcurementFactory(submission_id=model.submission_id + 1)
+    database.session.add_all(aps + [model, same_duns, unrelated])
+
+    monkeypatch.setattr(jobQueue.fileE, 'retrieveRows', Mock())
+    jobQueue.fileE.retrieveRows.return_value = [
+        fileE.Row('a', 'b', 'c', '1a', '1b', '2a', '2b', '3a', '3b',
+                  '4a', '4b', '5a', '5b'),
+        fileE.Row('A', 'B', 'C', '1A', '1B', '2A', '2B', '3A', '3B',
+                  '4A', '4B', '5A', '5B')
+    ]
+
+    # Mock out the interface holder class; rather nasty, as we want to _keep_
+    # the database session handler
+    interface_class = Mock()
+    interface_class.return_value.jobDb.session = database.session
+    jobQueue.generate_e_file(
+        model.submission_id, 1, interface_class, 'uniq', 'uniq',
+        is_local=True)
+    file_path = str(mock_broker_config_paths['broker_files'].join('uniq'))
+    expected = [
+        ['AwardeeOrRecipientUniqueIdentifier',
+         'UltimateParentUniqueIdentifier',
+         'UltimateParentLegalEntityName',
+         'HighCompOfficer1Name', 'HighCompOfficer1Amount',
+         'HighCompOfficer2Name', 'HighCompOfficer2Amount',
+         'HighCompOfficer3Name', 'HighCompOfficer3Amount',
+         'HighCompOfficer4Name', 'HighCompOfficer4Amount',
+         'HighCompOfficer5Name', 'HighCompOfficer5Amount'],
+        ['a', 'b', 'c', '1a', '1b', '2a', '2b', '3a', '3b', '4a', '4b',
+         '5a', '5b'],
+        ['A', 'B', 'C', '1A', '1B', '2A', '2B', '3A', '3B', '4A', '4B',
+         '5A', '5B']
+    ]
+    assert read_file_rows(file_path) == expected
```

Comment:
Indeed, this test is only meant to cover `generate_e_file`, which generates a CSV based on `fileE.Row`s. I'll see if there's not some more useful tests to write at a lower level, though.